File: js/insertionSort.js
const insertionSort = array => {
	for (let i = 0; i < array.length; i += 1) {
		let temp = array[i]
		let j = i - 1;
		while (j >= 0 && array[j] > temp) {
			array[j + 1] = array[j];
			j -= 1;
		}
		array[j + 1] = temp;
	}
	return array;
};

File: ruby/insertion_sort.rb
def insertion_sort(array)
	array.each_with_index do |num, index|
		temp = array[index]
		left = index - 1
		while left >= 0 && array[left] > temp
			array[left + 1] = array[left]
			left -= 1
		end
		array[left + 1] = temp
		print array
		puts
	end
end

array = [9,2,5,2,9,0]
puts insertion_sort(array)


